src/components/atoms/Timeline/index.tsx

```diff
@@ -0,0 +1,171 @@
+  // Convert memory count to pixel.
+  const currentPosition = useMemo(() => {
+    if (currentTime < startRange || currentTime > endRange) {
+      return 0;
+    }
+    const diff = (currentTime - startRange) / EPOCH_SEC / memoryInterval;
+    const currentMemoryCount = hoursCount * STRONG_MEMORY_HOURS;
+    return (
+      (diff * (currentMemoryCount * (GAP_HORIZONTAL + NORMAL_MEMORY_WIDTH) + STRONG_MEMORY_WIDTH)) /
+        currentMemoryCount +
+      GAP_HORIZONTAL / 2
+    );
+  }, [currentTime, startRange, hoursCount, endRange, memoryInterval]);
+
+  return (
+    <Container>
+      <MemoryList
+        startRange={startRangeDate}
+        memoryCount={memoryCount}
+        memoryInterval={memoryInterval}
+        hoursCount={hoursCount}
+      />
+      {/* TODO: Subtract icon width from position left. Currently Icon is displayed left corner as pointer. */}
+      <Icon style={{ left: currentPosition + PADDING_HORIZONTAL }}>{renderIcon()}</Icon>
+    </Container>
+  );
+};
+
+const Container = styled.div`
+  position: relative;
+  background: #303846;
+  overflow-x: auto;
+`;
+
+const Icon = styled.div`
+  position: absolute;
+  top: 2px;
+`;
+
+type MemoryListProps = {
+  startRange: Date;
+  memoryCount: number;
+  memoryInterval: number;
+  hoursCount: number;
+};
+
+const MemoryList: FC<MemoryListProps> = memo(
+  function MemoryListPresenter({ startRange, memoryCount, memoryInterval, hoursCount }) {
+    return (
+      <MemoryContainer>
+        {[...Array(memoryCount)].map((_, idx) => {
+          const isHour = idx % hoursCount === 0;
+          const isThreeHours = idx % (hoursCount * STRONG_MEMORY_HOURS) === 0;
+          if (isThreeHours) {
+            const d = new Date(startRange.getTime() + idx * EPOCH_SEC * memoryInterval);
+            const year = d.getFullYear();
+            const month = padStart(`${d.getMonth() + 1}`, 2, "0");
+            const date = padStart(`${d.getDate()}`, 2, "0");
+            const hour = padStart(`${d.getHours()}`, 2, "0");
+            // TODO: Fix date format
+            const label = `${year}/${month}/${date} ${hour}:00`;
+
+            return (
+              <LabeledMemory>
+                <MemoryLabel>{label}</MemoryLabel>
+                <Memory key={`${idx}`} isHour={isHour} isThreeHours={isThreeHours} />
+              </LabeledMemory>
+            );
+          }
+          return <Memory key={`${idx}`} isHour={isHour} isThreeHours={isThreeHours} />;
+        })}
+      </MemoryContainer>
+    );
+  },
+  (prev, next) => prev.memoryCount === next.memoryCount && prev.startRange === next.startRange,
+);
+
+const MemoryContainer = styled.div`
+  display: flex;
+  gap: 0 ${GAP_HORIZONTAL}px;
+  min-width: 100%;
+  height: 32px;
+  align-items: flex-end;
+  padding-left: ${PADDING_HORIZONTAL}px;
+  ::after {
+    content: "";
+    display: block;
+    padding-right: ${PADDING_HORIZONTAL}px;
+    height: 1px;
+  }
+`;
+
+type MemoryProps = {
```

Comment:
If this type is only used with the Memory component below, please keep them together. But also, with props so simple, we usually just write them in directly to styled components.
So below would be like
```
const Memory = styled.div<{isHour: boolean, isThreeHours: boolean}>`....`
```

Reply:
I fixed your review!
[fix: inline props type for styled](https://github.com/reearth/reearth-web/pull/256/commits/bc48dcde0f390e6c5a27a02736c2bd1b018c0501)